tests/test_docopt.py

```diff
@@ -487,40 +490,53 @@ def test_long_options_error_handling():
     #        docopt('Usage: prog --non-existent', '--non-existent')
     #    with raises(DocoptLanguageError):
     #        docopt('Usage: prog --non-existent')
-    with raises(DocoptExit):
+    with raises(
+        DocoptExit,
+        match=r"Warning: found unmatched \(duplicate\?\) arguments.*--non-existent",
+    ):
         docopt("Usage: prog", "--non-existent")
-    with raises(DocoptExit):
+    with raises(
+        DocoptExit, match=r"Warning: found unmatched \(duplicate\?\) arguments.*--ver\b"
+    ):
         docopt(
             "Usage: prog [--version --verbose]\n" "Options: --version\n --verbose",
             "--ver",
         )
-    with raises(DocoptLanguageError):
+    # --long is missing ARG in usage
+    with raises(DocoptLanguageError, match=r"unmatched '\('"):
         docopt("Usage: prog --long\nOptions: --long ARG")
-    with raises(DocoptExit):
+    with raises(DocoptExit, match=r"--long requires argument"):
         docopt("Usage: prog --long ARG\nOptions: --long ARG", "--long")
-    with raises(DocoptLanguageError):
+    with raises(DocoptLanguageError, match=r"--long must not have an argument"):
         docopt("Usage: prog --long=ARG\nOptions: --long")
-    with raises(DocoptExit):
+    with raises(DocoptExit, match=r"--long must not have an argument"):
         docopt("Usage: prog --long\nOptions: --long", "--long=ARG")
 
 
 def test_short_options_error_handling():
-    with raises(DocoptLanguageError):
+    with raises(DocoptLanguageError, match=r"-x is specified ambiguously 2 times"):
         docopt("Usage: prog -x\nOptions: -x  this\n -x  that")
 
-    with raises(DocoptExit):
+    with raises(
+        DocoptExit, match=r"Warning: found unmatched \(duplicate\?\) arguments.*-x"
+    ):
         docopt("Usage: prog", "-x")
 
-    with raises(DocoptLanguageError):
+    # -o needs an argument in usage. The "unmatched" error is because `-o` is
```

Comment:
Could you open a separate issue for this?

Reply:
Sure, I've taken out the message assertion/comment. It's a weird quirk of the way the parser is implemented, not at all intuitive. (Just to avoid any confusion, this isn't something I've changed here, it's just existing behaviour that I was asserting the message of.)

I've fixed the issue in a separate branch I'm working on to improve the usability of the error messages, it can wait for that.